### PR TITLE
boot: revert bb7d327a36d8fd923126052c32792e1293c1d243

### PR DESCRIPTION
**Pull request by valentindavid**
This reverts #13402.

`tests/nested/manual/uc-update-assets-secure-add-sbat` shows some failures from time to time. It is possible in some context, the wrong hashes are pruned when resealing.

### CODE REVIEW

**Comment by ernestl** [comment]
What about `seal_test.go`?

**Review by ernestl — APPROVED** [review]
Thanks

File: boot/seal_test.go
Comment (alfonsosanchezbeato) [inline boot/seal_test.go:1]: I see that this file changed since the commit this PR is reverting was merged, but this test fix is a bit strange. Was https://github.com/snapcore/snapd/pull/13402 fixing this issue? Was this a long standing problem?

Comment (valentindavid) [reply boot/seal_test.go:1]: No. It was not supposed to fix the issue. It assumed the issue did not exist. The issue is that for some reason, we get confused what is the new or and old grub binaries when preparing the boot chain to seal. I have no idea where the bug is at yet. But this is only in the sealing. The files get written correctly on the disk.

#13402 assumed that we knew which hash was which, that is the first hash is always the old one and the second one the new hash. But it appeared to be wrong. So that means #13402 did not work in that case.

The change that in this test that did not get reverted, changed something that made little sense to test. However I since this test got bigger and tested more case since #13402, I have added checks that we allow the old grub to be started from the new shim. But it is not really the old grub... it might be the new grub that we confused for the old grub. I know it is complicated.

I can also remove this test altogether, it passes, it is fine. `boot/bootchain_test.go` is enough. But I am not going to revert what was in `boot/seal_test.go`, because the test was completely changed, and what was removed made little sense to test.

Comment (alfonsosanchezbeato) [reply boot/seal_test.go:1]: I see, thanks for the explanation. So is this an issue that needs to be fixed before we merge https://github.com/snapcore/snapd/pull/13402 again, I guess? And after that in theory #13402 should be correct?

Approving now, but maybe it would be good to have a TODO to remember to change the test when the issue if fixed.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

File: boot/seal_test.go
Comment (valentindavid) [reply boot/seal_test.go:1]: I am working on it right now. But it safer to revert first.

**Pull request merged**
